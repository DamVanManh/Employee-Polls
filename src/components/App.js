import { useEffect } from "react";
import { connect } from "react-redux";
import { Routes, Route } from "react-router-dom";
import Navbar from "./Navbar";
import LoginPage from "./LoginPage";
import Dashboard from "./Dashboard";
import AddQuestion from "./AddQuestion";
import Leaderboard from "./Leaderboard";
import PollQuestion from "./PollQuestion";
import PageNotFound from "./PageNotFound";
import { setQuestions } from "../actions/questions";
import { setUsers } from "../actions/users";
import { _getUsers, _getQuestions } from "../utils/_DATA";

const App = ({ user, dispatch }) => {
  useEffect(() => {
    dispatch(async (dispatch) => {
      const { users, questions } = await Promise.all([
        _getUsers(),
        _getQuestions(),
      ]).then(([users, questions]) => ({
        users,
        questions,
      }));
      dispatch(setQuestions(questions));
      dispatch(setUsers(users));
    });
  }, []);
  return (
    <div className="container row d-flex justify-content-center m-auto">
      {user === null ? (
        <LoginPage />
      ) : (
        <>
          <Navbar />
          <Routes>
            <Route exact path="/" element={<Dashboard />} />
            <Route path="/leaderboard" element={<Leaderboard />} />
            <Route path="/add" element={<AddQuestion />} />
            <Route path="/questions/:question_id" element={<PollQuestion />} />
            <Route path="/pagenotfound" exact element={<PageNotFound />} />
            <Route path="*" element={<PageNotFound />} />
          </Routes>
        </>
      )}
    </div>
  );
};

const mapStateToProps = ({ user }) => ({
  user,
});

export default connect(mapStateToProps)(App);
