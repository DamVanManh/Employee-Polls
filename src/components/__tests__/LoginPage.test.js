import { createStore } from "redux";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import App from "../App";
import { users, questions } from "../../utils/mockdata";
describe("Test LoginPage", () => {
  it("Should match the snapshot", async () => {
    const storeMock = createStore(
      (state = { users, questions, user: null }) => state
    );
    jest.spyOn(storeMock, "dispatch").mockImplementation(() => {});

    const view = render(
      <Provider store={storeMock}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    expect(view).toMatchSnapshot();
  });

  it("Should show the LoginPage", async () => {
    const storeMock = createStore(
      (state = { users, questions, user: null }) => state
    );
    jest.spyOn(storeMock, "dispatch").mockImplementation(() => {});

    const view = render(
      <Provider store={storeMock}>
        <Router>
          <App />
        </Router>
      </Provider>
    );
    await view.findByText(/Login/i);
  });
});
