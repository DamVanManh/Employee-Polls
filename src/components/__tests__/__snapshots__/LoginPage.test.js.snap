// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginPage Should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container row d-flex justify-content-center m-auto"
      >
        <div
          class="justify-content-center d-flex align-items-center mt-5"
        >
          <form>
            <h1>
              Login
            </h1>
            <label>
              Select user:
            </label>
            <div
              class="row"
            >
              <div
                class=" p-3 hover rounded col-4"
                data-testid="sarahedo"
                data-user="sarahedo"
              >
                <img
                  alt="Sarah Edo"
                  class="avatar"
                  data-user="sarahedo"
                  src="/images/avatar1.jpeg"
                />
                <span
                  class="px-2"
                  data-user="sarahedo"
                >
                  Sarah Edo
                </span>
              </div>
              <div
                class=" p-3 hover rounded col-4"
                data-testid="tylermcginnis"
                data-user="tylermcginnis"
              >
                <img
                  alt="Tyler McGinnis"
                  class="avatar"
                  data-user="tylermcginnis"
                  src="/images/avatar2.jpeg"
                />
                <span
                  class="px-2"
                  data-user="tylermcginnis"
                >
                  Tyler McGinnis
                </span>
              </div>
              <div
                class=" p-3 hover rounded col-4"
                data-testid="mtsamis"
                data-user="mtsamis"
              >
                <img
                  alt="Mike Tsamis"
                  class="avatar"
                  data-user="mtsamis"
                  src="/images/avatar3.jpeg"
                />
                <span
                  class="px-2"
                  data-user="mtsamis"
                >
                  Mike Tsamis
                </span>
              </div>
              <div
                class=" p-3 hover rounded col-4"
                data-testid="zoshikanlu"
                data-user="zoshikanlu"
              >
                <img
                  alt="Zenobia Oshikanlu"
                  class="avatar"
                  data-user="zoshikanlu"
                  src="/images/avatar4.jpeg"
                />
                <span
                  class="px-2"
                  data-user="zoshikanlu"
                >
                  Zenobia Oshikanlu
                </span>
              </div>
            </div>
            <div
              class="text-center"
            >
              <button
                class="btn btn-primary"
                data-testid="Submit"
                disabled=""
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container row d-flex justify-content-center m-auto"
    >
      <div
        class="justify-content-center d-flex align-items-center mt-5"
      >
        <form>
          <h1>
            Login
          </h1>
          <label>
            Select user:
          </label>
          <div
            class="row"
          >
            <div
              class=" p-3 hover rounded col-4"
              data-testid="sarahedo"
              data-user="sarahedo"
            >
              <img
                alt="Sarah Edo"
                class="avatar"
                data-user="sarahedo"
                src="/images/avatar1.jpeg"
              />
              <span
                class="px-2"
                data-user="sarahedo"
              >
                Sarah Edo
              </span>
            </div>
            <div
              class=" p-3 hover rounded col-4"
              data-testid="tylermcginnis"
              data-user="tylermcginnis"
            >
              <img
                alt="Tyler McGinnis"
                class="avatar"
                data-user="tylermcginnis"
                src="/images/avatar2.jpeg"
              />
              <span
                class="px-2"
                data-user="tylermcginnis"
              >
                Tyler McGinnis
              </span>
            </div>
            <div
              class=" p-3 hover rounded col-4"
              data-testid="mtsamis"
              data-user="mtsamis"
            >
              <img
                alt="Mike Tsamis"
                class="avatar"
                data-user="mtsamis"
                src="/images/avatar3.jpeg"
              />
              <span
                class="px-2"
                data-user="mtsamis"
              >
                Mike Tsamis
              </span>
            </div>
            <div
              class=" p-3 hover rounded col-4"
              data-testid="zoshikanlu"
              data-user="zoshikanlu"
            >
              <img
                alt="Zenobia Oshikanlu"
                class="avatar"
                data-user="zoshikanlu"
                src="/images/avatar4.jpeg"
              />
              <span
                class="px-2"
                data-user="zoshikanlu"
              >
                Zenobia Oshikanlu
              </span>
            </div>
          </div>
          <div
            class="text-center"
          >
            <button
              class="btn btn-primary"
              data-testid="Submit"
              disabled=""
              type="submit"
            >
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
